code/datums/ai_laws.dm

```diff
@@ -5,16 +5,48 @@
 #define LAW_ION "ion"
 
 
+// ai_laws holds lists of laws
+// Each law has a description of actual law telling the player what to do
+// and also has a state for storing whether the law should be stated pubicly or not
+// state can be changed the the player
+
+// description is the actual law
+// state is whether the law should be stated publicly or not (AI's choice)
+/datum/ai_law
+	var/description = null
+	var/state = null
+
+/datum/ai_law/New(law_description, lawState = TRUE)
+	description = law_description
+	state = lawState
+
 /datum/ai_laws
 	var/name = "Unknown Laws"
-	var/zeroth = null
-	var/zeroth_borg = null
+	var/id = DEFAULT_AI_LAWID
+	var/mob/living/silicon/owner
+
+	// Lists of laws
 	var/list/inherent = list()
 	var/list/supplied = list()
 	var/list/ion = list()
-	var/mob/living/silicon/owner
-	var/list/devillaws = null
-	var/id = DEFAULT_AI_LAWID
+	// devil laws are multiple laws given to a borg as borgs can be devils
+	var/list/devil_laws = null

```

Comment:
You either need to preinitialize this list like ion, inherent, and supplied are above or lazy initialize it as-needed. Some of the helpers below assume the list is always instanced when it isn't.